### PR TITLE
6.3 new methods

### PR DESCRIPTION
**Pull request by Poolitzer**
I am keeping the branch updated with the merge to one :D

Currently implemented:
 - [x] get forum topic icon stickers
 - [x] edit forum topic
 - [x] create forum topic
 - [x] close forum topic
 - [x] delete forum topic
 - [x] reopen forum topic
 - [x] unpin all topic messages

### CODE REVIEW

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8397]: this should be added in extbot as well

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8475]:
```suggestion
                :tg-const:`telegram.constants.TopicLimit.MIN_TITLE_LENGTH`-
                :tg-const:`telegram.constants.TopicLimit.MAX_TITLE_LENGTH` characters.
```

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8477]:
```suggestion
                the topic icon. Use :meth:`~telegram.Bot.get_forum_topic_icon_stickers` to get all
```
since we're already in bot, let's shorten it

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8712]: also editForumTopic to be added

File: tests/test_forum.py
Comment (harshil21) [inline tests/test_forum.py:28]:
```suggestion
```

File: tests/test_forum.py
Comment (harshil21) [inline tests/test_forum.py:44]: well, now we do :D

**Review by harshil21 — COMMENTED** [review]
initial review

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8542]:
```suggestion
                of the target channel (in the format ``@supergroupusername``).
```

File: telegram/_bot.py
Comment (harshil21) [inline telegram/_bot.py:8469]:
```suggestion
                of the target channel (in the format ``@supergroupusername``).
```

File: telegram/_message.py
Comment (harshil21) [inline telegram/_message.py:2828]:
```suggestion
            await bot.reopen_forum_topic(
```

**Review by harshil21 — COMMENTED** [review]
review 2 (fixing this myself)

File: telegram/_bot.py
Comment (Bibo-Joshi) [inline telegram/_bot.py:8518]: please add a todo comment about `ForumTopic.de_json` missing here

File: telegram/_bot.py
Comment (Bibo-Joshi) [inline telegram/_bot.py:8563]: .

File: telegram/_bot.py
Comment (Bibo-Joshi) [inline telegram/_bot.py:8821]: .

File: telegram/_bot.py
Comment (Bibo-Joshi) [inline telegram/_bot.py:8472]: `.. versionadded: 20.0` is missing for all methods

File: tests/test_message.py
Comment (Bibo-Joshi) [inline tests/test_message.py:69]: please also a `message_thread_id` entry to `message_params` fixture below

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]

File: telegram/_bot.py
Comment (harshil21) [reply telegram/_bot.py:8432]: This is only done in `doc-fixes` iisc, and we've also started using rst substitutions for some params, so have to do that after API PRs are merged ig

Comment (Bibo-Joshi) [reply telegram/_bot.py:8432]: ah, right … I guess it's time to merge the doc-fixes PR - there's a lot of stuff in there already

Comment (Bibo-Joshi) [reply telegram/_bot.py:6910]: > time to merge the doc-fixes PR

done

**Comment by Bibo-Joshi** [comment]
IMO we can merge, then. We'll have to do a final review of #3360 anyway :D

**Comment by lemontree210** [comment]
I'll mention #3340 here if you don't mind. Helps my mental navigation :)

**Comment by lemontree210** [comment]
I know it's merged already so I'm late, but it looks like the params `icon_color` and `icon_custom_emoji_id` of method `Bot.create_forum_topic` are optional in [Telegram API](https://core.telegram.org/bots/api#createforumtopic) but aren't optional in [our `bot.py`](https://github.com/python-telegram-bot/python-telegram-bot/pull/3360/files#diff-2aef7cabe736b556967d73b7f946c32a67142d6bccb4c90a355d375374ebd8eaR6918-R6919)?

**Pull request merged**
